ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/ContractStore.scala

```diff
@@ -26,43 +26,20 @@ trait ContractStore {
       loggingContext: LoggingContext
   ): Future[Option[ContractId]]
 
-  /** This method serves two purposes:
-    *   1 - Verify that none of the specified contracts are archived (archival of divulged contracts also count)
-    *   2 - Calculate the maximum ledger time of all the specified contracts (divulged contracts do not contribute)
-    * Important note: existence of the contracts is not checked, only the fact of archival.
-    *
-    * @return NotAvailable, if none of the specified contracts are archived, and all of them are divulged contracts (no ledger-time available)
-    *         NotAvailable, if the specified set is empty.
-    *         Max, if none of the specified contracts are archived, and the maximum ledger-time of all specified non-divulged contracts is known
-    *         Archived, if there was at least one contract specified which is archived (this list is not necessarily exhaustive)
-    */
-  def lookupMaximumLedgerTimeAfterInterpretation(ids: Set[ContractId])(implicit
-      loggingContext: LoggingContext
-  ): Future[MaximumLedgerTime]
-
-  def lookupContractForValidation(
+  def lookupContractStateWithoutDivulgence(
```

Comment:
As this appears to be nuanced I some scaladoc would be good.  For me, provided it is well documented, `lookupContractState` would be a nicer name.